cp23/hpi-fun2/motivation/problem.tex

```diff
@@ -0,0 +1,49 @@
+\input{template.tex}
+
+\begin{document}
+
+\makeheader
+
+One day, Christian, the CEO of \textit{Super Arrangement Productions}, noticed that his company
+was not running as he wanted. His employees complained about the lack of motivation.
+Therefore, Christian decided to hire the motivation agency \textit{KcMinsey} to send their best
+coaches in order to motivate the people again.
+
+The coaches are taught to always work the same way. First, all coaches are sent to one boss to
+motivate them, starting with Christian himself. Since they are insecure, they always work together,
+which does not increase their power to motivate.
+Then, they separate and motivate the employees below that boss. Since the coaches are a bit lazy,
+they do not help their colleagues when they reached the lowest level of the hierarchy.
+
+Sadly, motivating people is very exhausting. That means, that each person trying to motivate
+an employee suffers in their motivational level equally to the demotivation level an employee
+has. But after that, employees are as motivated as their coach was after the talk. Then,
+they will even support their coaches and become coaches themselves.
+
+Some people are not motivatable at all. Probably, because \textit{KcMinsesy}'s coaches
+are too exhausted when they reach them. That would also mean not being able to motivate the
+employees below that person anymore.
```

Comment:
Ok sorry, now I'm not sure anymore that I grasp the problem correctly. Based on your formalization you gave in the issue, that is how I currently understand the problem:
1. We are given a directed rooted tree.
2. We want to cover the tree with directed paths (which corresponds to coaches descending the company hierarchy)
3. Each path can start anywhere (or only at the root?)
4. Each path has a maximal allowed weight i. e. the sum of it's vertex weights
5. Find the minimum number of paths to cover the tree 
Is this correct?

Reply:
I designed the solution to hold the following:
3. Paths can only start at the root.
4. Indeed, with the addition, that it is part of the path can exceed the weight (most likely you mean the same thing).
5. Indeed, but each node "increases the number of coaches". So you would take the number of paths to leafs (a leaf is a node, that has no reachable children) and subtract the number of inner nodes (answer must be at least 1).

Reply:
I imagine the algorithm to work like this (should I include this image in the description?):

![test drawio](https://github.com/compprog-lecture-tools/fun-contest-2023/assets/50635122/f0338fcb-fbc2-4833-99d6-96a13af5f980)

Reply:
Visuals are always good idea, but you maybe want to use tikz for that job or create a more polished version of your illustration for the problem statement. Before I'll try to recap again, could you explain the case of a node with 3 children?

Reply:
Sure thing! Also here I assume, that we do not end up in an unreachable leaf.

![test drawio-2](https://github.com/compprog-lecture-tools/fun-contest-2023/assets/50635122/29c30cd3-c527-42ce-a7db-28a17535bace)

Reply:
Thank you! And lastly, could you give a quick example with weights?

Reply:
This is the example with weights: I marked already what the algorithm needs to recognize in order to find the solution:

![test drawio-3](https://github.com/compprog-lecture-tools/fun-contest-2023/assets/50635122/6169d08b-039d-4b97-8df4-764ec3e16cdb)

Reply:
Great! Now I get it! Note that your limits might be a bit too loose. The limits currently allow quadratic solutions. I'd recommend a limit of 10^5 nodes.

Reply:
I'll write my solution and see if this works out.

Reply:
You are right. That should be solvable with 10^5. I will change that.